modules/asset-manager-api/src/main/java/org/opencastproject/assetmanager/api/fn/Product.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package org.opencastproject.assetmanager.api.fn;
+
+public final class Product<A> {
```

Comment:
What does this do and why do we need this? This is just a generic wrapper around one object. Why do we need to wrap something in here?

Reply:
I don't really know what this does, which is exactly why we need this unfortunately :D.

This is a class from the entwine library, stripped down to the bare minimum we actually use. I could not figure out how to replace this with "proper" Java code. To see how it is used take a look at `assetmanager/api/Value.java`.